guillotina/contrib/dbusers/subscribers.py

```diff
@@ -44,3 +44,18 @@
         if group.id not in user.user_groups:
             user.user_groups.append(group.id)
             user.register()
+
+
+@configure.subscriber(for_=(IUser, IObjectModifiedEvent))
+async def on_update_user(user: User, event: ObjectModifiedEvent) -> None:
+    # Keep group.users and user.user_groups in sync
+    container = get_current_container()
+    user_groups = user.user_groups or []
+    for group_id in user_groups:
+        try:
+            group = await navigate_to(container, f"groups/{group_id}")
+        except KeyError:
+            raise HTTPPreconditionFailed(content={"reason": f"inexistent group: {group_id}"})
+        if user.id not in group.users:
+            group.users.append(user.id)
+            group.register()
```

Comment:
I think its missing the logic for when a group is removed in the `user_groups`. IMO would be better to implement the `IObjectBeforeModifiedEvent` and get the differences between `user.user_groups` and `payload["user_groups"]`. That way we know which groups are added and which ones are deleted.

Reply:
nice idea!
Anywyay there is something that feels like so bad... 
On the current PR, why the test fails when runs in conjuntion with test_manage_groups.py, 
There is somehting contaminated on it.

Reply:
o my god! I'm hitting an issue that smells like a mutable shared list.. :/

Reply:
This might be the problem: https://github.com/plone/guillotina/blob/241524086b35a016e214c0d0b8b6908ea047f586/guillotina/contrib/dbusers/content/groups.py#L26

I think it should be `defaultFactory=list` instead of `default=[]`

Reply:
Also this: https://github.com/plone/guillotina/blob/241524086b35a016e214c0d0b8b6908ea047f586/guillotina/contrib/dbusers/content/groups.py#L38